### ISSUE TITLE
PAL: Cleanup GetModuleFileName** and implement corresponding interface

### ISSUE THREAD

**Issue by obastemur** [Microsoft/ChakraCore#2297]
This is a follow-up issue for #2296 

```
+// xplat-todo: Implement a corresponding solution for GetModuleFileNameW
+// and cleanup PAL. [ https://github.com/Microsoft/ChakraCore/pull/2288 should be merged first ]
+// GetModuleFileName* PAL is not reliable and forces us to explicitly double initialize PAL
+// with argc / argv....
```

~~Waiting for #2288~~

Move `GetBinaryLocation` (https://github.com/Microsoft/ChakraCore/blob/7541e228ef42a97895a88f46d5427d07bf2675ba/bin/ch/Helpers.cpp#L577) into PlatformAgnostic and use that instead of `GetModuleFileName*`

edit-2, Some more details:

- Create `SystemInfo.cpp` under `lib/Runtime/PlatformAgnostic/Platform/Common/`

```
#include "Common.h"
#include "ChakraPlatform.h"

namespace PlatformAgnostic
{
    void SystemInfo::GetBinaryLocation(........) {   .......  }
    ....
```

- After moving GetBinaryLocation under the new SystemInfo.cpp, we should add it into `lib/Runtime/PlatformAgnostic/CMakeLists.txt` (i.e. at line 4)

- Similarly add that source file into `lib/Runtime/PlatformAgnostic/Chakra.Runtime.PlatformAgnostic.vcxproj` (i.e. line 41)

- Finally add SystemInfo's GetBinaryLocation static method definition into `lib/Common/PlatformAgnostic/SystemInfo.h`

**Comment by dilijev** [comment]
@obastemur Any update on this?

**Comment by obastemur** [comment]
@dilijev this is a nice to have hence not rushing on. Also added `accepting PRs` label. If anyone is interested implementing this, highly appreciated.

**Comment by obastemur** [comment]
#3112 takes care of moving GetBinaryLocation into PlatformAgnostic. 

Remaining task: remove GetModuleFileName** interface and replace the usages to GetBinaryLocation.
